android/src/main/java/com/reactnativeteadssdkmodule/ArrayUtil.kt

```diff
@@ -0,0 +1,113 @@
+package com.reactnativeteadssdkmodule
+
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableType
+import com.facebook.react.bridge.WritableArray
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+
+/*
+  ArrayUtil exposes a set of helper methods for working with
+  ReadableArray (by React Native), Object[], and JSONArray.
+  MIT License
+  Copyright (c) 2020 Marc Mendiola
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+  The above copyright notice and this permission notice shall be included in all
+  copies or substantial portions of the Software.
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+  SOFTWARE.
+ */
+
+
+
+object ArrayUtil {
```

Comment:
just for the record (u don't need to change it) in case you don't know a nice way to have this kind of function in koltin is by using extension functions: https://kotlinlang.org/docs/extensions.html#extensions-are-resolved-statically

for instance
```
  fun ReadableArray.toJSONArray(): JSONArray {
    val jsonArray = JSONArray()
    for (i in 0 until this.size()) {
      val type = this.getType(i)
      when (type) {
        ReadableType.Null -> jsonArray.put(i, null)
        ReadableType.Boolean -> jsonArray.put(i, this.getBoolean(i))
        ReadableType.Number -> jsonArray.put(i, this.getDouble(i))
        ReadableType.String -> jsonArray.put(i, this.getString(i))
        ReadableType.Map -> jsonArray.put(i, MapUtil.toJSONObject(this.getMap(i)))
        ReadableType.Array -> jsonArray.put(i, toJSONArray(this.getArray(i)))
      }
    }
    return jsonArray
  }
```